tests_e2e/orchestrator/lib/agent_test_suite_combinator.py

```diff
@@ -218,36 +218,44 @@ def create_environment_list(self) -> List[Dict[str, Any]]:
                 else:
                     vm_size = ""
 
-                if suite_info.owns_vm:
-                    # create an environment for exclusive use by this suite
-                    environment_list.append({
+                # Note: Disabling "W0640: Cell variable 'foo' defined in loop (cell-var-from-loop)". This is a false positive, the closure is OK
+                # to use, since create_environment() is called within the same iteration of the loop.
+                # pylint: disable=W0640
+                def create_environment(env_name: str) -> Dict[str, Any]:
+                    tags = {}
+                    if suite_info.template != '':
+                        tags["templates"] = suite_info.template
+                    return {
                         "c_marketplace_image": marketplace_image,
                         "c_cloud": self.runbook.cloud,
                         "c_location": location,
                         "c_vm_size": vm_size,
                         "c_vhd": vhd,
                         "c_test_suites": [suite_info],
-                        "c_env_name": f"{name}-{suite_info.name}",
+                        "c_env_name": env_name,
                         "c_marketplace_image_information_location": self._MARKETPLACE_IMAGE_INFORMATION_LOCATIONS[self.runbook.cloud],
-                        "c_shared_resource_group_location": self._SHARED_RESOURCE_GROUP_LOCATIONS[self.runbook.cloud]
-                    })
+                        "c_shared_resource_group_location": self._SHARED_RESOURCE_GROUP_LOCATIONS[self.runbook.cloud],
+                        "c_vm_tags": tags
```

Comment:
Introduced a new variable, "c_vm_tags", which is used to pass the custom templates to the UpdateArmTemplateHook (see wiki for details)